packages/components/tooltip/src/tooltip.tsx

```diff
@@ -244,21 +258,33 @@ const Tooltip = (props: TTooltipProps) => {
         onBlur(event);
       }
 
-      if (closeAfter) {
+      if (closeAfter && state === 'opened') {
         leaveTimer.current = setTimeout(() => {
-          handleClose(event);
+          const tooltipElement = popperInstance?.popper.querySelector(
+            '[data-testid="tooltip-message-wrapper"]'
+          ) as HTMLElement;
+          tooltipElement.addEventListener(
+            'animationend',
+            (event: AnimationEvent) => {
+              const element = event.target as HTMLElement;
+              element.style.display = 'none';
```

Comment:
This is something I'm now sure about.
After the transition is done, the tooltip gets visible again and I don't understand why, so I just hide it before it gets removed in the next tick.

Reply:
if we add some basic logging of the current state on each rerender
```js
useEffect(() => {
    console.log(state);
  });
```
it seems that there is some inconsistency of state changes.
<img width="179" alt="Screenshot 2023-07-24 at 15 17 05" src="https://github.com/commercetools/ui-kit/assets/49066275/57329cfe-1341-43f5-8ce1-df03ee141de5">
I think changing `setState('exiting');` to `setState('closed');` solves the issue with the tooltip popping up again (but I don't think that's what we expect).

Reply:
Thanks for looking into it, Kacper.

I don't seem to be able to reproduce that behaviour.

If you can share any context I can use that would be helpful.

Also, if you can see why the events are in that unexpected order, that would be great.

Reply:
Thanks Carlos, and sorry for not being precise.

Perhaps this brings no value, it's just my observation derived from your comment. 

The incorrect order (and the effect of the tooltip showing up 2 times) happens when we eliminate a part of the code, where you added the `popperInstance` to hide the tooltip with css.

If we omit the `exiting` state in favor of `closed` like in the diff below it works without any manual changes in the css (the sequence is `entering` -> `opened` -> `closed`).

<img width="568" alt="Screenshot 2023-07-25 at 08 48 04" src="https://github.com/commercetools/ui-kit/assets/49066275/984109cd-585d-4f41-a0ab-0301c2eb2281">

Please let me know if you need any more info.

Reply:
Thanks for the explanation, Kacper 👍 
But if we get rid of the "exiting" state, I think we lose the new required close transition.
We now need to show and hide the tooltip with a transition. If we don't wait for the transition to end before setting the state to "close", the tooltip will be removed from the DOM before the transition ends.

Reply:
Anyway, I removed (9687276a119f9408be21c6f1acac0ce7622c6744) the `display` _hack_ and everything is working fine.

Maybe it was something I added while I was working in the feature and with the final implementation is not needed.